h2o-jetty-8/src/main/java/water/webserver/jetty8/Jetty8ServerAdapter.java

```diff
@@ -0,0 +1,118 @@
+package water.webserver.jetty8;
+
+import org.eclipse.jetty.server.Handler;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.server.handler.HandlerCollection;
+import org.eclipse.jetty.server.handler.HandlerWrapper;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import water.webserver.iface.H2OHttpView;
+import water.webserver.iface.RequestAuthExtension;
+import water.webserver.iface.WebServer;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+class Jetty8ServerAdapter implements WebServer {
+  private final Jetty8Helper helper;
+  private final H2OHttpView h2oHttpView;
+  private Server jettyServer;
+
+  private Jetty8ServerAdapter(Jetty8Helper helper, H2OHttpView h2oHttpView) {
+    this.helper = helper;
+    this.h2oHttpView = h2oHttpView;
+  }
+
+  static WebServer create(final H2OHttpView h2oHttpView) {
+    final Jetty8Helper helper = new Jetty8Helper(h2oHttpView);
+    return new Jetty8ServerAdapter(helper, h2oHttpView);
+  }
+
+  @Override
+  public void start(String ip, int port) throws Exception {
+    jettyServer = helper.createJettyServer(ip, port);
+    final HandlerWrapper handlerWrapper = helper.authWrapper(jettyServer);
+    final ServletContextHandler context = helper.createServletContextHandler();
+    registerHandlers(handlerWrapper, context);
+    jettyServer.start();
+  }
+
+  /**
+   * Stop Jetty server after it has been started.
+   * This is unlikely to ever be called by H2O until H2O supports graceful shutdown.
+   *
+   * @throws Exception -
+   */
+  @Override
+  public void stop() throws Exception {
+    if (jettyServer != null) {
+      jettyServer.stop();
+    }
+  }
+
+  private void registerHandlers(HandlerWrapper handlerWrapper, ServletContextHandler context) {
+    for (Map.Entry<String, Class<? extends HttpServlet>> entry : h2oHttpView.getServlets().entrySet()) {
+      context.addServlet(entry.getValue(), entry.getKey());
+    }
+
+    final List<Handler> extHandlers = new ArrayList<>();
+    extHandlers.add(helper.authenticationHandler());
+    // here we wrap generic authentication handlers into jetty-aware wrappers
+    final Collection<RequestAuthExtension> authExtensions = h2oHttpView.getAuthExtensions();
+    for (final RequestAuthExtension requestAuthExtension : authExtensions) {
+      extHandlers.add(new AbstractHandler() {
+        @Override
+        public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+          if (requestAuthExtension.handle(target, request, response)) {
```

Comment:
```java
baseRequest.setHandled(requestAuthExtension.handle(target, request, response));
```

Reply:
That would slightly change the semantic (probably unimportant).
But I find it better to have one method per line, especially if one is non-trivial - better for stacktraces, debugging, readability.